sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_storage_utils.py

```diff
@@ -140,6 +157,8 @@ def get_storage_client(
         account_url = STORAGE_ACCOUNT_URLS[storage_type].format(storage_account, storage_endpoint)
 
     if storage_type == DatastoreType.AZURE_BLOB:
+        if use_async:
+            return AsyncBlobStorageClient(credential=credential, container_name=container_name, account_url=account_url)
```

Comment:
I've only made the changes for Blob so far - so the same would need to be done for DataLake etc.